datafusion/core/src/physical_plan/coalesce_partitions.rs

```diff
@@ -137,27 +131,17 @@ impl ExecutionPlan for CoalescePartitionsExec {
                 // use a stream that allows each sender to put in at
                 // least one result in an attempt to maximize
                 // parallelism.
-                let (sender, receiver) =
-                    mpsc::channel::<Result<RecordBatch>>(input_partitions);
+                let mut builder =
```

Comment:
I am quite pleased that this is now all encapsulated into `RecordBatchReceiverStream`